src/compiler/checker.ts

```diff
@@ -7092,6 +7082,7 @@ namespace ts {
                     case SyntaxKind.Identifier:
                     case SyntaxKind.PropertyAccessExpression:
                     case SyntaxKind.QualifiedName:
+                        // TODO (drosen): Why a qualified name?

```

Comment:
It may not be needed - I believe it was just for completeness, since `EntityName` (which is what the function on the following line takes) is `Identifier | QualifiedName` - provided I'm remembering that right.

It may very well be that the grammar prohibits a `QualifiedName` from being passed to this check.